src/xwayland/xwm/mod.rs

```diff
@@ -284,6 +309,25 @@ pub trait XwmHandler {
     ///
     /// The window will be holding a grab on the mouse button provided.
     fn move_request(&mut self, xwm: XwmId, window: X11Surface, button: u32);
+
+    /// Window requests access to the given selection.
+    fn allow_selection_access(&mut self, xwm: XwmId, selection: SelectionType) -> bool {
+        let _ = (xwm, selection);
+        false
+    }
+
+    /// The given selection is being read by an X client and needs to be written to the provided file descriptor
+    fn send_selection(&mut self, xwm: XwmId, selection: SelectionType, mime_type: String, fd: OwnedFd);
+
+    /// A new selection was set by an X client with provided mime_types
+    fn new_selection(&mut self, xwm: XwmId, selection: SelectionType, mime_types: Vec<String>) {
+        let _ = (xwm, selection, mime_types);
+    }
+
+    /// A proviously set selection of an X client got cleared
+    fn cleared_selection(&mut self, xwm: XwmId, selection: SelectionType) {
+        let _ = (xwm, selection);
+    }
```

Comment:
I feels weird to have most of those methods be optional with a default of doing nothing, but not all. Feels inconsistent to me.

Reply:
Well, it is a balance of operations you need to implement to provide a somewhat usable Xwm and those that could be skipped (like clipboard synchronization). Its a narrow line for sure. Any recommendations?

Reply:
I probably don't know enough about X11 selection, but my first understanding would be that if `allow_selection_access` always returns `false`, then `send_selection` would never be invoked? And so why require it to be implemented?

Probably that's not what happens though, but in that case maybe it'd be worth expanding the documentation to explain, for the optional methods, what behavior you get if you don't implement them.

Reply:
> I probably don't know enough about X11 selection, but my first understanding would be that if allow_selection_access always returns false, then send_selection would never be invoked? And so why require it to be implemented?

No that is exactly right and a good hint. I'll give `send_selection` a default impl, that panics. (Because it needs to be handled once you override allow_selection_access.)